src/main/java/com/epam/reportportal/cucumber/AbstractReporter.java

```diff
@@ -44,19 +61,26 @@
  * @author Serhii Zharskyi
  * @author Vitaliy Tsvihun
  */
-public abstract class AbstractReporter implements Formatter {
+public abstract class AbstractReporter implements ConcurrentEventListener {
 
+    Supplier<Launch> launch;
+    static final String COLON_INFIX = ": ";
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractReporter.class);
 
-    protected static final String COLON_INFIX = ": ";
+    private Map<Pair<String, String>, RunningContext.FeatureContext> currentFeatureContextMap =
+            new HashMap<Pair<String, String>, RunningContext.FeatureContext>();
 
-    /* feature context */
-    protected RunningContext.FeatureContext currentFeatureContext;
+    private Map<Pair<String, String>, RunningContext.ScenarioContext> currentScenarioContextMap =
```

Comment:
i would suggest making key as simple object (string, for example) instead of complex one

Reply:
Could you please suggest a better way of storing scenario Name and feature URI.
Can it be like values separated by any symbol like ":" or something like that?